### PR TITLE
Notify log out/in rather than force logout

### PR DESCRIPTION
**Pull request by daved**
https://www.pivotaltracker.com/story/show/176928819

### CODE REVIEW

**Review by Naatan — COMMENTED** [review]
I like the idea of adding a tip here, but the tips themselves are meant to be suuuper short and to the point, eg. "Logout with `state auth logout`".

**Comment by daved** [comment]
@Naatan This will produce an effect only in some cases since we generally reach this code using `authentication.Get()`. In other words, the error w/tip will not propagate in most cases. I'm not sure how else to resolve this without touch many file.

**Comment by Naatan** [comment]
> @Naatan This will produce an effect only in some cases since we generally reach this code using `authentication.Get()`. In other words, the error w/tip will not propagate in most cases. I'm not sure how else to resolve this without touch many file.

That's fine. Removing `auth.Get()` is its own story.

File: pkg/platform/authentication/auth.go [hunk 1]
```diff
@@ -140,18 +140,23 @@ func (s *Auth) Authenticate() error {
...
 	params.SetCredentials(credentials)
-	loginOK, err := mono.Get().Authentication.PostLogin(params)
 
+	loginOK, err := mono.Get().Authentication.PostLogin(params)
 	if err != nil {
-		s.Logout()
+		tips := []string{
+			locale.T("relog_tip", "If you're seemingly logged in already, try logging out and in again."),
+			locale.T("logout_tip", "Logout with[ACTIONABLE]`state auth logout`[/RESET]."),
+			locale.T("logout_tip", "Login with[ACTIONABLE]`state auth`[/RESET]."),
```
Comment (Naatan) [inline pkg/platform/authentication/auth.go:149]:
```suggestion
			locale.T("logout_tip", "Logout with [ACTIONABLE]`state auth logout`[/RESET]."),
			locale.T("logout_tip", "Login with [ACTIONABLE]`state auth`[/RESET]."),
```

File: pkg/platform/authentication/auth.go [hunk 2]
```diff
@@ -140,18 +140,23 @@ func (s *Auth) Authenticate() error {
 func (s *Auth) AuthenticateWithModel(credentials *mono_models.Credentials) error {
 	params := authentication.NewPostLoginParams()
 	params.SetCredentials(credentials)
-	loginOK, err := mono.Get().Authentication.PostLogin(params)
 
+	loginOK, err := mono.Get().Authentication.PostLogin(params)
 	if err != nil {
-		s.Logout()
+		tips := []string{
+			locale.T("relog_tip", "If you're seemingly logged in already, try logging out and in again."),
```
Comment (Naatan) [inline pkg/platform/authentication/auth.go:147]:
```suggestion
			locale.T("relog_tip", "If you're having trouble authenticating try logging out and logging back in again."),
```

**Review by Naatan — CHANGES_REQUESTED** [review]

**Review by Naatan — APPROVED** [review]

**Pull request merged**